EIPS/eip-681.md

```diff
@@ -0,0 +1,78 @@
+## Preamble
+
+    EIP: 681
+    Title: URL Format for Transaction Requests
+    Author: Daniel A. Nagy <daniel@ethereum.org>
+    Type: Standard Track
+    Category: ERC
+    Status: Draft
+    Replaces: 67
+    Created: 2017-08-01
+    Requires: 20, 137
+
+## Simple Summary
+A standard way of representing various transactions, especially payment requests in Ethers and ERC #20 tokens as URLs.
+
+## Abstract
+URLs embedded in QR-codes, hyperlinks in web-pages, emails or chat messages provide for robust cross-application signaling between very loosely coupled applications. A standardized URL format for payment requests allows for instant invocation of the user's preferred wallet application (even if it is a webapp or a swarm đapp), with the correct parameterization of the payment transaction only to be confirmed by the (authenticated) user.
+
+## Motivation
+The convenience of representing payment requests by standard URLs has been a major factor in the wide adoption of Bitcoin. Bringing a similarly convenient mechanism to Ethereum would speed up its acceptance as a payment platform among end-users. In particular, URLs embedded in broadcast Intents are the preferred way of launching applications on the Android operating system and work across practically all applications. Desktop web browsers have a standardized way of defining protocol handlers for URLs with specific protocol specifications. Other desktop applications typically launch the web browser upon encountering a URL. Thus, payment request URLs could be delivered through a very broad, ever growing selection of channels.
+
+This specification supersedes ERC #67, which is a URL format for representing arbitrary transactions in a low-level fashion. This ERC focuses specifically on the important special case of payment requests, while allowing for other, ABI-specified transactions.
+
+## Specification
+
+### Syntax
+Payment request URLs contain "ethereum" in their schema (protocol) part and are constructed as follows:
+
+    request                 = "ethereum" ":" target_address [ "@" chain_id ] [ "/" function_name ] [ "?" parameters ]
+    target_address          = ethereum_address
+    chain_id                = 1*DIGIT
+    function_name           = STRING
+    ethereum_address        = ( "0x" 40*40HEXDIG ) / ENS_NAME
+    parameters              = parameter *( "&" parameter )
+    parameter               = key "=" value
+    key                     = "value" / "gas" / TYPE
```

Comment:
Just noticing that this document never states that gas is the gas-price not the limit. Perhaps this should be stated. Just noticed after seeing this: https://github.com/walleth/walleth/issues/102#issue-273610986

Reply:
This absolutely needs to be distinguished. Coding preliminary URL formats myself has always necessitated `gasPrice` and `gasLimit` identifiers. `gas` really has no semantic meaning.

Reply:
On that note, it does feel like `gasLimit` should be added to the list of possible values for `key`.  Consider adding `nonce` as well so these URLs car be used for replacement transactions all well.